### PR TITLE
fix(sdk): summarize sdk component crashes with stage index errors

### LINKED ISSUE #50398: SDK crashes with "Stage 1 does not exist" when Summarize component is rendered

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/50398

### Description

Fixes the stage index error thrown when rendering the [Summarize](https://github.com/metabase/metabase/pull/49901) sdk component. The only consumers of `SummarizeAggregationItemList` and `SummarizeBreakoutColumnList` used in the SDK's `Summarize` component is `SummarizeSidebar`, which is only used by `StructuredQueryRightSidebar` which only passes `stageIndex={-1}`. This makes sense as we only want to work with the latest stages in this case.

### How to verify

- Go to an interactive question's default view (or a view with the Summarize view)
- Click on the Summarize button
- It should not crash. The layout does look weird but this will be fixed in https://github.com/metabase/metabase/issues/49719

### Checklist

- [x] Tests have been added/updated to cover changes in this PR - E2E

### CODE REVIEW

**Review by oisincoveney — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Summarize.tsx [hunk 1]
```diff
@@ -55,22 +52,22 @@ const SummarizeInner = ({
     onClose();
   };
 
-  const hasAggregations = Lib.aggregations(currentQuery, stageIndex).length > 0;
+  const hasAggregations = Lib.aggregations(currentQuery, -1).length > 0;
```
Comment (uladzimirdev) [inline enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Summarize.tsx:55]: nit: create a var `stageIndex` and pass it instead

File: e2e/test/scenarios/embedding-sdk/interactive-question.cy.spec.ts [hunk 2]
```diff
@@ -173,4 +173,24 @@ describeSDK("scenarios > embedding-sdk > interactive-question", () => {
     getSdkRoot().contains("onBeforeSave is called");
     getSdkRoot().contains("question saved as Sample Orders 4");
   });
+
+  it("should not crash when clicking on Summarize", () => {
```
Comment (kamilmielnik) [inline e2e/test/scenarios/embedding-sdk/interactive-question.cy.spec.ts:177]:
```suggestion
  it("should not crash when clicking on Summarize (metabase#50398)", () => {
```

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Summarize.tsx (hunk 1 above)
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Summarize.tsx]: Thanks! Fixed in https://github.com/metabase/metabase/pull/50400/commits/c2c41882b2124dc552dbcc68d10aa4098da292b5

File: e2e/test/scenarios/embedding-sdk/interactive-question.cy.spec.ts (hunk 2 above)
Comment (heypoom) [reply e2e/test/scenarios/embedding-sdk/interactive-question.cy.spec.ts]: Thanks!

**Pull request merged**